src/net_processing.cpp

```diff
@@ -4717,7 +4712,7 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
         // to unreasonably increase our timeout.
         if (state.vBlocksInFlight.size() > 0) {
             QueuedBlock &queuedBlock = state.vBlocksInFlight.front();
-            int nOtherPeersWithValidatedDownloads = nPeersWithValidatedDownloads - (state.nBlocksInFlightValidHeaders > 0);
```

Comment:
So I guess here, since we know `state.vBlocksInFlight.size() > 0` you're able to get rid of this condition in the subtraction: `state.nBlocksInFlightValidHeaders > 0`?

Is it always true that `state.vBlocksInFlight.size()` == `state->nBlocksInFlight`?

Reply:
Along these lines, is `mapBlocksInFlight` and `vBlocksInFlight` tracking the same underlying state, just as different data structures? I think the ordering of `vBlocksInFlight` is important though?

Reply:
> Is it always true that state.vBlocksInFlight.size() == state->nBlocksInFlight?

Yes. There are only two places where `vBlocksInFlight` is updated, and `nBlocksInFlight` is updated at the same time. if `vBlocksInFlight.size()` was not the same as `nBlocksInFlight`, it'd be a bug.

> is mapBlocksInFlight and vBlocksInFlight tracking the same underlying state, just as different data structures? I think the ordering of vBlocksInFlight is important though?

`mapBlocksInFlight` is a global data structure, and holds iterators into the various `vBlocksInFlight` for the different peers. So you can think of the `vBlocksInFlight` lists as tracking the state, and `mapBlocksInFlight` as an index into that state.

Eventually, I'd like to extract all of the block request tracking logic and state into its own submodule like the txrequest module, which was added in #19988. A boost multi-index might be the right way to track this state if we did that.

Reply:
Thanks for the explanations - makes sense.

> Eventually, I'd like to extract all of the block request tracking logic and state into its own submodule like the txrequest module, which was added in #19988. A boost multi-index might be the right way to track this state if we did that.

Sounds great, I was trying to figure out if something like this can/should be done with these questions and seems like you're already thinking about it 👍

Anyways very nice cleanup in this PR, code reads much better now